src/System.Management.Automation/engine/SpecialVariables.cs

```diff
@@ -289,6 +293,7 @@ internal static class SpecialVariables
                                                                     /* WarningPreference */     typeof(ActionPreference),
                                                                     /* InformationPreference */ typeof(ActionPreference),
                                                                     /* ConfirmPreference */     typeof(ConfirmImpact),
+                                                                    /* PSDefaultFileEncoding */ typeof(Microsoft.PowerShell.FileEncoding),
```

Comment:
This line should be removed. I realize it's not obvious why this array exists, but it is used for variables that are always created, and these preference variables get created in every script cmdlet, but your new preference variable should not.